tests/lib/Value/ItemListTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Personalization\Value;
+
+use Ibexa\Contracts\Personalization\Value\ItemInterface;
+use Ibexa\Contracts\Personalization\Value\ItemListInterface;
+use Ibexa\Personalization\Value\Storage\ItemList;
+use Ibexa\Tests\Personalization\Creator\DataSourceTestItemCreator;
+use Ibexa\Tests\Personalization\Storage\AbstractDataSourceTestCase;
+use OutOfBoundsException;
+
+/**
+ * @covers \Ibexa\Personalization\Value\Storage\ItemList
+ */
+final class ItemListTest extends AbstractDataSourceTestCase
+{
+    public function testThrowExceptionWhenFirstElementNotExists(): void
+    {
+        $this->expectException(OutOfBoundsException::class);
+        $this->expectExceptionMessage('Collection is empty');
+
+        $itemList = new ItemList([]);
+        $itemList->first();
```

Comment:
If exception for some reason would be thrown in the constructor, this test would stop testing what it should. Not that it will happen in this case, but the PHPUnit syntax was changed from annotation to method, so you can place exception expectation right before it is expected to happen:
```suggestion
        $itemList = new ItemList([]);
        
        $this->expectException(OutOfBoundsException::class);
        $this->expectExceptionMessage('Collection is empty');
        
        $itemList->first();
```